ouroboros-network-framework/src/Simulation/Network/Snocket.hs

```diff
@@ -721,13 +737,23 @@ mkSnocket state tr = Snocket { getLocalAddr
           -- accepted.
           FDUninitialised mbLocalAddr -> mask $ \unmask -> do
             (connId, bearerInfo, simOpen) <- atomically $ do
-              bearerInfo <- stepScriptSTM (nsBearerInfo state)
               localAddress <-
                 case mbLocalAddr of
                   Just addr -> return addr
                   Nothing   -> nsNextEphemeralAddr state (getAddressType remoteAddress)
               let connId = ConnectionId { localAddress, remoteAddress }
 
+              attenuationMap <- readTVar (nsAttenuationMap state)
+
+              bearerInfo <- case Map.lookup connId attenuationMap of
+                Nothing     -> do
+                  script <- stepScriptSTMTx (nsBearerInfo state)
+                  writeTVar (nsAttenuationMap state)
+                            (Map.insert connId script attenuationMap)
```

Comment:
This defies the purpose of having an explicit `Map ConnectionId BearerInfo`, drawing things from a list will not improve shrinking (we're already doing that).

Reply:
fixed!